docs/book/v1/intro.md

```diff
@@ -42,6 +40,37 @@ if a user has been authenticated or not, e.g. it can be used to verify the
 authorization level of a user (for this scope, it is consumed by
 [zend-expressive-authorization](https://github.com/zendframework/zend-expressive-authorization)).
 
+## Default User class
+
+We provide a default implementation of `UserInterface` via the class `Zend\Expressive\Authentication\DefaultUser`.
+This class is a basic implementation of `UserInterface`. This class is final and
+it's immutable, that means you cannot change it state at runtime.
+In order to set the identity and the user's role we provided a default factory
+class that generates a `UserInterface` object. This factory is
+`Zend\Expressive\Authentication\UserInterfaceFactory`. This class uses a `generate`
+function to create a `UserInterface` instance passing the identity and the roles
+(if any) of the user.
+
+If you want, you can customize the `UserInterfaceFactory` using your custom
+`UserInterface` implementation. You need to change the service configuration as
+follows:
+
+```php
+// src/ConfigProvider.php
```

Comment:
I would demonstrate this as straight configuration, and not as part of a `ConfigProvider`. The reasons:

- Application developers will likely want to know how to override the configuration at the application level, which is in the `config/autoload/*.php` files.
- Package developers (those creating adapters) will know that the configuration goes in a `ConfigProvider` class.